### PR TITLE
chore(doc): document using Smart Triggers

### PR DESCRIPTION
**Pull request by aali309**
fixes: #143
Added documentation on Using Smart Triggers

### CODE REVIEW

**Comment by aali309** [comment]
Not sure if I added a lot.. Please let me know if I need to add or chop off some of the stuff.

File: guides/_subsections/using-smart-triggers.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+## [Using Smart Triggers](#using-smart-triggers)
+Cryostat 2.4 agent supports custom triggers that are based on MBean metric values. You can configure Cryostat agent to start JFR recordings dynamically when these custom trigger conditions are met.
```
Comment (andrewazores) [inline guides/_subsections/using-smart-triggers.md:2]: We don't need to include versioning information directly in the text here - that's one of the benefits of versioning the entire guides directory :)

File: guides/_subsections/using-smart-triggers.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+You can set up a custom trigger condition in Cryostat to initiate Java Flight Recorder (JFR) recordings dynamically. This trigger condition is based on MBean counters, covering various runtime, memory, thread, and OS metrics. The condition includes MBean counter types and allows specifying a duration for the trigger to activate only if the specified values persist for that duration. Cryostat's agent supports smart triggers, which continuously monitor MBean counter values. When the current values match the configured conditions for the specified duration, the trigger initiates a JFR recording dynamically through the Cryostat agent.
+
+**NOTE:** A JFR recording will not start dynamically if the custom trigger condition associated with this recording is not met.
+
+#### [Configure Custom Trigger for Dynamic Recording](#configure-custom-trigger-for-dynamic-recording)
+When you configure your target application to load the Cryostat agent, you can define one or more custom triggers that are then passed as arguments to the agent.
+For more information see [Using-the-Cryostat-Agent](#using-the-cryostat-agent).
```
Comment (andrewazores) [inline guides/_subsections/using-smart-triggers.md:10]: The rendered text here shouldn't have the `-` hyphens in it:

image

File: guides/_subsections/using-smart-triggers.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+You can define a custom trigger in any of the following ways:
+
+<ol>
+  <li>
+    <details>
+      <summary>Appending a Custom Trigger to the Cryostat Agent’s JAR File Path</summary>
+      <p>
+The following example shows how to append a simple custom trigger to the Cryostat agent’s JAR file path:
+      </p>
+      <pre>
```
Comment (andrewazores) [inline guides/_subsections/using-smart-triggers.md:22]: All these `<pre>` blocks end up formatted a bit strangely:

image

The text block is scrollable to the right, and the indentation seems unnecessary.

File: guides/_subsections/using-smart-triggers.md (hunk 2 above)
Comment (aali309) [reply guides/_subsections/using-smart-triggers.md:10]: ooh yes :)

Comment (andrewazores) [reply guides/_subsections/using-smart-triggers.md]: I think this should be the other way around, and maybe we can rephrase it a little:

```
For more information, learn how to [use the Cryostat Agent](#using-the-cryostat-agent).
```

File: guides/_subsections/using-smart-triggers.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    <li>The name of the event template must be defined after the trigger expression and preceded by a tilde <code>(~)</code> character. For example: <pre>~profile</pre></li>
+    <li>A trigger expression can consist of one or more constraints and a target duration. The set of constraints and target duration must be separated by a semicolon <code>(;)</code> character.</li>
+    <li>Each constraint must include: the name of an MBean counter; a relational operator such as <code> ></code> (greater than), <code>=</code> (equal to), <code> < </code>(less than), and so on; and a specified value. The type of relational operator and value that you can specify depends on the associated MBean counter type. For example: <pre>ProcessCpuLoad > 0.2</pre></li>
+    <li>Constraints can be grouped together by using logical operators such as <code>&&</code> (AND), <code>|| </code> (OR), or <code> ! </code> (NOT) logic. For readability and clarity around the order of operations and operator precedence, grouped constraints may be enclosed in round brackets, but this is not a requirement. For example:
```
Comment (andrewazores) [inline guides/_subsections/using-smart-triggers.md:69]: Here there are some spaces captured inside the code blocks, like `<code>|| </code>` - I think this should just be `<code>||</code>`?

**Review by andrewazores — APPROVED** [review]

**Pull request merged**